### PR TITLE
Bump to NLPModelsJuMP 0.13 (and JuMP new nonlinear interface)

### PR DESCRIPTION
**Pull request by tmigot**
Hi @paraynaud ! I tried to update, to more recent JuMP versions but ran into this issue:
```
get_expression_tree: Error During Test at .julia\dev\ExpressionTreeForge.jl\test\NLPSupport.jl:5
  Got exception outside of a @test
  Unable to query objective_expr because no nonlinear objective was set
  Stacktrace:
    [1] error(s::String)
      @ Base .\error.jl:35
    [2] objective_expr
      @ .julia\packages\MathOptInterface\1fRdT\src\Nonlinear\evaluator.jl:95 [inlined]
    [3] get_expression_tree(model::Model)
      @ ExpressionTreeForge .julia\dev\ExpressionTreeForge.jl\src\NLPSupport\get_expr_tree.jl:16
    [4] macro expansion
      @ .julia\dev\ExpressionTreeForge.jl\test\NLPSupport.jl:11 [inlined]
    [5] macro expansion
      @ AppData\Local\Programs\Julia-1.10.4\share\julia\stdlib\v1.10\Test\src\Test.jl:1577 [inlined]
    [6] top-level scope
      @ .julia\dev\ExpressionTreeForge.jl\test\NLPSupport.jl:6
    [7] include(fname::String)
      @ Base.MainInclude .\client.jl:489
    [8] top-level scope
      @ .julia\dev\ExpressionTreeForge.jl\test\runtests.jl:10
    [9] include(fname::String)
      @ Base.MainInclude .\client.jl:489
   [10] top-level scope
      @ none:6
   [11] eval
      @ .\boot.jl:385 [inlined]
   [12] exec_options(opts::Base.JLOptions)
      @ Base .\client.jl:291
   [13] _start()
      @ Base .\client.jl:552
Test Summary:       | Error  Total  Time
get_expression_tree |     1      1  6.1s
```
Would you have some time to have a look?

### CODE REVIEW

**Comment by paraynaud** [comment]
Hello @tmigot, I failed to fix the error for now, but I know where it comes from.
The good news is that it does not come from MathOptNLPModel, but from JuMP.Model directly.

The tests fail on this part
```julia
n = 10
jump_model = PureJuMP.arwhead(; n)
nlp_mo = MathOptNLPModel(jump_model)
nlp_ad = ADNLPProblems.arwhead(; n)

expr_jump = get_expression_tree(jump_model) # fails
expr_mo = get_expression_tree(nlp_mo) # works
expr_ad = get_expression_tree(nlp_ad) # works
```

The issue is that
```julia
jump_model.nlp_model == nothing
```
or equivalently
```julia
nlp = JuMP.nonlinear_model(jump_model; force=true)
nlp.objective == nothing
```
Consequently, 
```julia
evaluator = MOI.Nonlinear.Evaluator(nlp)
MathOptInterface.initialize(evaluator, [:ExprGraph])
obj_Expr = MathOptInterface.objective_expr(evaluator)::Expr # fails
```
fails as it considers that no objective is defined.

I don't know why, but the use of `MathOptNLPModel(jump_model)` solves the issue, as an objective is properly set.
Do you have an idea on how NLPModelsJuMP manages to make it happen ?

**Comment by tmigot** [comment]
Would something like this work for you?
```
using OptimizationProblems, JuMP, MathOptInterface

n = 10
model = PureJuMP.arwhead(; n)
nlp = JuMP.nonlinear_model(jump_model; force=true)
evaluator = MOI.Nonlinear.Evaluator(nlp)
F = MOI.get(model, MOI.ObjectiveFunctionType())
MOI.get(model, MOI.ObjectiveFunction{F}())
if F <: MOI.ScalarNonlinearFunction
    MOI.Nonlinear.set_objective(nlp, MOI.get(model, MOI.ObjectiveFunction{F}()))
    has_nonlinear = true
end
obj_Expr = MathOptInterface.objective_expr(evaluator)

```
inspired by https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl/blob/a5d1aae9110b23f4043adf3168e259a45f27d122/src/utils.jl#L462

**Comment by paraynaud** [comment]
Thanks @tmigot, I couldn't ask for more.
Feel free to rebase if you're satisfied (and release a new version if needed).

File: src/NLPSupport/get_expr_tree.jl [hunk 1]
```diff
@@ -3,27 +3,32 @@ using Symbolics
...
-  expr_tree =
-    ExpressionTreeForge.transform_to_expr_tree(obj_Expr)::ExpressionTreeForge.Type_expr_tree
-  return expr_tree
+  nlp = JuMP.nonlinear_model(model; force=true)
+  evaluator = MOI.Nonlinear.Evaluator(nlp)
+  F = MOI.get(model, MOI.ObjectiveFunctionType())
+  MOI.get(model, MOI.ObjectiveFunction{F}())
+  if F <: MOI.ScalarNonlinearFunction
+      MOI.Nonlinear.set_objective(nlp, MOI.get(model, MOI.ObjectiveFunction{F}()))
+      # has_nonlinear = true
```
Comment (tmigot) [inline src/NLPSupport/get_expr_tree.jl]:
```suggestion
```

**Comment by tmigot** [comment]
Thanks a lot @paraynaud for the help!

**Pull request merged**